frontends/smv_encoder.cpp

```diff
@@ -1,2 +1,86 @@
 #include "smv_encoder.h"
 
+using namespace smt;
+using namespace pono;
+using namespace std;
+
+int pono::SMVEncoder::parse(std::string filename)
+{
+  std::ifstream ifs;
+  ifs.open(filename);
+  if (!ifs.good()) {
+    std::cout << "NO input file!" << std::endl;
+    exit(-1);
+  }
+  std::istream & s = ifs;
+  pono::SMVscanner smvscanner(*this);
+  smvscanner.switch_streams(&s);
+  pono::smvparser parse(smvscanner, *this);
+  return parse();
+}
+// parse string
+int pono::SMVEncoder::parseString(std::string newline)
```

Comment:
If it's not too difficult, it would be really cool if `parseString` could return a `Term` instead of directly adding it to the transition system. That way you can do something like `Term xp1 = smv_encoder.parseString("x + 0ud4_1");`